### DISCUSSION TITLE
difference between model and model ema

### DISCUSSION THREAD

**Discussion by MohamedDonia** [ultralytics/yolov5#5844] [Q&A]
in training , there are model used for training  and model ema used in validation step, what is the difference between then ??
and why they not output the same dimensions ??

**Comment by glenn-jocher** [comment]
@MohamedDonia for EMA explanation see https://www.tensorflow.org/api_docs/python/tf/train/ExponentialMovingAverage

**Reply by MohamedDonia** [reply]
sorry, i can not figure out why model get output of shape (nb, 3, 80, 80 , num_classes+5) while ema get output of shape (nb, predictions , 6) @glenn-jocher
